worker_specific_task_queues/worker_specific_activities.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'digest'
+require 'logger'
+require 'net/http'
+require 'tempfile'
+require 'temporalio/activity'
+require 'uri'
+
+module WorkerSpecificTaskQueues
+  module Activities
+    class DownloadFileActivity < Temporalio::Activity::Definition
+      def execute(url)
+        # Simulate slow activity
+        sleep(3)
+
+        # Download file using block syntax for automatic cleanup
+        temp_path = nil
+        Tempfile.open do |file|
+          Temporalio::Activity::Context.current.logger.info("Downloading #{url} to #{file.path}")
+          file.write(Net::HTTP.get(URI(url)))
+          temp_path = file.path
+        end
+
+        temp_path
```

Comment:
```suggestion
          file.path
        end
```

Return of block is same as return of call